services/config/src/useDate.ts

```diff
@@ -0,0 +1,134 @@
+import { useCallback, useContext, useMemo } from 'react'
+import { ConfigContext } from './ConfigContext'
+import { DateComponents, DateInput } from './types'
+
+// extend date with extra methods
+class DHIS2Date extends Date {
+    serverOffset: number
+    serverTimezone: string
+    clientTimezone: string
+
+    constructor({
+        date,
+        serverOffset,
+        serverTimezone,
+        clientTimezone,
+    }: {
+        date: DateInput
+        serverOffset: number
+        serverTimezone: string
+        clientTimezone: string
+    }) {
+        if (date) {
+            super(date)
+        } else {
+            super()
+        }
+        this.serverOffset = serverOffset
+        this.serverTimezone = serverTimezone
+        this.clientTimezone = clientTimezone
+        Object.setPrototypeOf(this, DHIS2Date.prototype)
```

Comment:
What's the purpose of this? Shouldn't the prototype of an instance of the `DHIS2Date` class already have this prototype?

Reply:
I think this should be explained in a comment

Reply:
I think this was is just left-over from a different implementation that I had before. I'll get rid of it 🙏